documentation/src/test/java/example/extensions/ParameterResolverNestedClasses.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example.extensions;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+// tag::user_guide[]
+public class ParameterResolverNestedClasses {
+
+	@Test
+	@DisplayName("Outer class test")
+	void testOuterClass(TestInfo testInfo) {
+		assertEquals("Outer class test", testInfo.getDisplayName());
+	}
+
+	@Nested
+	class FirstResolution {
+
+		@RegisterExtension
+		final FirstIntegerResolver firstIntegerResolver = new FirstIntegerResolver();
```

Comment:
Using `@ExtendWith` would be more idiomatic here and below because there's no state or configuration that would justify programmatic extension registration.

Reply:
Changed to `@ExtendWith` here and also in the other code examples.